src/codec.ts

```diff
@@ -358,6 +456,23 @@ function decode(value: Value, type: spannerClient.spanner.v1.Type): Value {
     case 'BYTES':
       decoded = Buffer.from(decoded, 'base64');
       break;
+    case spannerClient.spanner.v1.TypeCode.PROTO:
+    case 'PROTO':
+      decoded = Buffer.from(decoded, 'base64');
+      decoded = new ProtoMessage({
+        value: decoded,
+        fullName: type.protoTypeFqn,
+        messageFunction: columnInfo as Function,
```

Comment:
Add a check here and in enum

Reply:
Sorry my bad, the line isn't descriptive. In protos, column info is used as Function and inEnums it is an object. Can we check if this is possible(in case user passes an incorrect value) and gracefully throw an error

Reply:
We don't need to throw an error, so if the `columnInfo ` is missing then still data would be deserialised as bytes and similar to integer in case of enum.

Reply:
What I meant was what if it is present but in the wrong format

Reply:
In that case it will throw an error because deserialisation would fail.